src/components/Api.js

```diff
@@ -0,0 +1,126 @@
+export default class Api {
+  constructor(options) {
+    this._baseUrl = options.baseUrl;
+    this._headers = options.headers;
+  }
+
+  getUserProfileData() {
+    // returns a Promise type object which is "thenable" to continue the async chain to go forward once the JSON object is retrieved from the server via the API call
+    return fetch(`${this._baseUrl}/users/me`, {
+      headers: this._headers,
+    })
+      .then((res) => {
+        if (res.ok) return res.json(); // if successful, returns a user object
+        return Promise.reject(`error ${res.status} while getting user data`); // otherwise, output an error to .catch
+      })
+      .catch((err) => {
+        return Promise.reject(err); // properly return a rejected Promise state
+      });
```

Comment:
You should **not** use `catch` in `Api` because you need to `catch` possible errors at the very **end** of any request, and this end is in `index.js`  after all `then` blocks. And in real projects the `catch` blocks usually open error popups and so on.